backend/oqtopus_cloud/provider/routers/jobs.py

```diff
@@ -39,42 +40,73 @@
+) -> list[Union[JobDef, GetJobsResponse]] | ErrorResponse:
     logger.info("invoked get_jobs")
-    query = db.query(Job).filter(Job.device_id == device_id)
-    if status is not None:
-        query = query.filter(Job.status == status)
-    if timestamp is not None:
-        time = datetime.fromisoformat(timestamp).astimezone(jst)
-        query = query.filter(Job.created_at > time)
-    if max_results is not None:
-        query = query.limit(max_results)
-    models = query.all()
-    jobs: list[JobDef] = []
-    for model in models:
-        job = model_to_schema(model)
-        if isinstance(job, ValueError):
-            logger.warning(str(job))
+    # Fields Control
+    try:
+        if fields is not None:
+            fields_list = fields.split(",")
+            MAP_SCHEMA_TO_MODEL = {v: k for k, v in MAP_MODEL_TO_SCHEMA.items()}
+            converted_fields_list = [
+                MAP_SCHEMA_TO_MODEL[field] for field in fields_list
+            ]
+            columns = [getattr(Job, field) for field in converted_fields_list]
+            arg_select = columns
+        else:
+            arg_select = [Job]
+
+        stmt = select(*arg_select).filter(Job.device_id == device_id)
+        full_stmt = select(Job).filter(Job.device_id == device_id)
+        # query = db.query(Job).filter(Job.device_id == device_id)
+        # Filtering Jobs
+        if status is not None:
+            stmt = stmt.filter(Job.status == status)
+            full_stmt = full_stmt.filter(Job.status == status)
+        if timestamp is not None:
+            time = datetime.fromisoformat(timestamp).astimezone(jst)
+            stmt = stmt.filter(Job.created_at > time)
+            full_stmt = full_stmt.filter(Job.created_at > time)
+        if max_results is not None:
+            stmt = stmt.limit(max_results)
+            full_stmt = full_stmt.limit(max_results)
+
+        if fields is None:
+            # models is Job type
+            models = db.scalars(stmt).all()
         else:
-            try:
-                if decode_job_status(model.status) == JobStatus.submitted:
-                    model.status = JobStatus.ready
-                    db.commit()
-                    job.status = JobStatus(model.status)
-                jobs.append(job)
-            except Exception as e:
+            # models is Row type
+            models = db.execute(stmt).all()
+        update_models = db.scalars(full_stmt).all()
+        results: list[Union[JobDef, GetJobsResponse]] = []
+        for model, update_model in zip(models, update_models):
+            job = model_to_schema(model)
+            if isinstance(job, ValueError):
                 logger.warning(str(job))
-                logger.warning(f"Error: {str(e)}")
-    return jobs
+            else:
+                try:
+                    if decode_job_status(update_model.status) == JobStatus.submitted:
+                        update_model.status = JobStatus.ready
+                        db.commit()
```

Comment:
Please avoid `committ`ing inside the loop.

Reply:
https://github.com/oqtopus-team/oqtopus-cloud/blob/95b0788414394d812526f434c26ecd95285cd793/backend/oqtopus_cloud/provider/routers/jobs.py#L100-L113

Move commit to out of loop